docs/upgrade.html

```diff
@@ -142,6 +142,8 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                         </li>
                         <li>The deprecated <code>sendOffsetsToTransaction(Map&lt;TopicPartition, OffsetAndMetadata&gt;, String)</code> method has been removed from the Producer API.
                         </li>
+                        <li>The <code>flush</code> method now includes deadlock detection when invoked inside a callback to prevent unintended blocking scenarios.
```

Comment:
This could be more explicit, like the message in the exception added to `KafkaProducer`. You are not allowed to use `flush()` in the callback. Maybe "The flush method now includes deadlock detection preventing its use inside a callback. This avoids unintended blocking which has been known to occur prior to this change." Something like that. We want someone reading this to understand that they must not use flush in the callback.